### PR TITLE
[Dev 7.0] make dropdown options dynamic

### PR DESCRIPTION
**Pull request by xennex22**

### CODE REVIEW

File: src/lv_widgets/lv_dropdown.c [hunk 1]
```diff
@@ -195,7 +200,113 @@ void lv_dropdown_set_options(lv_obj_t * ddlist, const char * options)
...
+    ext->options = (char *)options;
+}
+
+/**
+ * Add an options to a drop down list from a string.  Only works for dynamic options.
+ * @param ddlist pointer to drop down list object
+ * @param options a string without '\n'. E.g. "Four"
+ * @param position the insert position, indexed from 0, -1 = end of string
+ */
+void lv_dropdown_add_option(lv_obj_t * ddlist, const char * option, int pos)
```
Comment (kisvegabor) [inline src/lv_widgets/lv_dropdown.c]: Please use `int16_t` or `int32_t` instead of `int`.

File: src/lv_widgets/lv_dropdown.c [hunk 2]
```diff
@@ -195,7 +200,113 @@ void lv_dropdown_set_options(lv_obj_t * ddlist, const char * options)
...
+
+    /*Can not append to static options*/
+    if(ext->static_txt != 0) return;
+
+    /*Allocate space for the new option*/
+    size_t old_len = (ext->options == NULL) ? 0 : strlen(ext->options);
+	size_t ins_len = strlen(option);
+	size_t new_len = ins_len + old_len + 2; /* +2 for terminating NULL and possible \n */
+	lv_mem_monitor_t mem_mon;
+	lv_mem_monitor(&mem_mon);
```
Comment (kisvegabor) [inline src/lv_widgets/lv_dropdown.c]: Why do you perform memory monitoring and defrag here?

**Review by kisvegabor — COMMENTED** [review]
Looks good in general. Thank you!
 I have only a few comments.

File: src/lv_widgets/lv_dropdown.c (hunk 1 above)
Comment (xennex22) [reply src/lv_widgets/lv_dropdown.c]: Ok, done.

File: src/lv_widgets/lv_dropdown.c (hunk 2 above)
Comment (xennex22) [reply src/lv_widgets/lv_dropdown.c]: Sorry, debug code left in place.

**Comment by kisvegabor** [comment]
All looks good, thank you!

**Pull request merged**